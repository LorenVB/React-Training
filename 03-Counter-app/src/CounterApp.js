import React, { Fragment, useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value = 10 }) => {
  const [counter, setCounter] = useState(value); //[2 items]

  // handleAdd
  const handleAdd = (e) => {
    setCounter(counter + 1);
  };

  const handleReset = (e) => {
    setCounter(value);
  };

  const handleSubstract = (e) => {
    setCounter(counter - 1);
  };

  return (
    <Fragment>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>

      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleSubstract}>-1</button>
    </Fragment>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

export default CounterApp;
