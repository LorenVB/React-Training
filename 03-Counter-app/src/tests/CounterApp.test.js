import React from "react";
import CounterApp from "../CounterApp";
const { shallow } = require("enzyme");

describe("Testing CounterApp />", () => {
    let wrapper;

    beforeEach(()=>{
         wrapper = shallow( < CounterApp / > );
    })

            test("should return <CounterApp /> correctly", () => {
               
                expect(wrapper).toMatchSnapshot();
            });

            test("should return default value 100", () => {
                    const wrapper = shallow( < CounterApp value = { 100 }
                        />);
                        const counterText = wrapper.find("h2").text(); 
                        
                        expect(counterText).toBe('100');
                    });

            test('should increment with +1 button', () => {
                wrapper.find('button').at(0).simulate('click');

                const counterText= wrapper.find('h2').text().trim();

                expect(counterText).toBe('11');
               
            })
            test('should increment with -1 button', () => {
                wrapper.find('button').at(2).simulate('click');
                const counterText= wrapper.find('h2').text().trim();

                expect(counterText).toBe('9');
               
            })

            test('should reset value to 10', () => {
              
                wrapper.find('button').at(0).simulate('click');
                wrapper.find('button').at(1).simulate('click');

                const counterText= wrapper.find('h2').text().trim();
                expect(counterText).toBe('10');
            })
            
            
            });