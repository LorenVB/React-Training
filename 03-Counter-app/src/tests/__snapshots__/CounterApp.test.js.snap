// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CounterApp /> should return <CounterApp /> correctly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
